### PR TITLE
For #22692: Remove Metropolis font.

### LINKED ISSUE #22692: Update Android to using System Fonts only
Hi team,

We want to shift towards using system fonts only, which means we'd like to change any existing custom fonts to the native font. From what I'm aware of, it is just the Metropolis font, please let me know if there are other any other fonts to consider. Would like to know if there are any implications or design requirements needed for this change.

Thanks!

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-18816)

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

Fixes #22692
Fixes #22692
Fixes #22692
Fixes #22692
Fixes #22692
Fixes #22692
Fixes #22692
Fixes #22692
Fixes #22692
Fixes #22692

### CODE REVIEW

**Comment by mcarare** [comment]
@topotropic This is the PR we talked about for removing the font.
Also cc: @cwzilla.

**Comment by cwzilla** [comment]
Hi team,

Copying over my comment from the original issue ---- Can we actually pause on this for now? I realize it won't be a simple change as updating the font, but we would need to audit and see where the current headline styles are being used and how that would actually translate to the system font and ensuring it still works with our UI and is accessible. I'll need to do a bit more research, I may have some further questions later. Thank you!

@mcarare

**Comment by gabrielluong** [comment]
I am wondering if we should get a UX review for this prior to moving forward or wait until the beginning of the next release?

File: app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt:188]: nit extra new line

File: app/src/main/res/layout/history_list_item.xml
Comment (gabrielluong) [inline app/src/main/res/layout/history_list_item.xml:9]: Seems weirdly ordered, can we move this with the rest of the xmlns

File: app/src/main/res/values/integers.xml
Comment (gabrielluong) [inline app/src/main/res/values/integers.xml:10]: nit add a new line for EOF

File: app/src/main/res/values/styles.xml
Comment (gabrielluong) [inline app/src/main/res/values/styles.xml:184]: nit let's leave this new line to separate the styles

**Comment by mcarare** [comment]
> @mcarare I am wondering if we should get a UX review (some screenshots for comparisons, apk testing) for this prior to moving forward or wait until the beginning of the next release? I am leaning toward the latter since soft freeze is on Thursdday and beta cut is less than a week.

@gabrielluong The changes were made using UX specs, so I am not sure it needs additional approval. We could wait to land this after the beta cut.

**Comment by gabrielluong** [comment]
> > @mcarare I am wondering if we should get a UX review (some screenshots for comparisons, apk testing) for this prior to moving forward or wait until the beginning of the next release? I am leaning toward the latter since soft freeze is on Thursdday and beta cut is less than a week.
> 
> @gabrielluong The changes were made using UX specs, so I am not sure it needs additional approval. We could wait to land this after the beta cut.

Let's wait until after the beta cut then.

File: app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:32]: Curious shouldn't this be `FontWeight.SemiBold` to match?

Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:32]: Nevermind, I see the comments for this. I am thinking we should post some screenshots for @cwzilla to just super confirm the changes by showing what the header looks like with `FontWeight.SemiBold` and `FontWeight.Medium`

Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:32]: Metropolis Semibold|Medium|Semibold
---|---|---
Screen Shot 2022-02-03 at 12 25 01 PM|Screen Shot 2022-02-03 at 12 23 50 PM|Screen Shot 2022-02-03 at 12 24 38 PM

**Comment by cwzilla** [comment]
Hi, thanks @gabrielluong for the screenshots.

It doesn't look quite right. Based on the Roboto specs [here](https://fonts.google.com/specimen/Roboto), for Medium 500 weight, it looks much bolder than in the screenshots.

Can we change the link type style to Body (so it should just appear regular).

**Comment by cwzilla** [comment]
@gabrielluong @mcarare  I think in addition, something that may help with the review is if dev can provide the current specs for each use case (ie. the homepage headers, etc.), since I provided the new specs based on what we have in Figma (which isn't always accurate to the app) and there may have been discrepancies.

**Comment by mcarare** [comment]
@cwzilla I am happy to make any changes based on the Figma links to the components that previously used the Metropolis font and now have the desired styling. 
So please point me to the place I can find them, otherwise, I feel like we are only going around in circles with this.

File: app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:32]: I just realized this was a misunderstanding. The screenshots are of the recently visited header, not of a `SectionHeader`.

Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:32]: Oh derp, you're right. What I have screenshot is Header16TextStyle in `styles.xml`. I think issue still lies on the fact that the "Jump back in", "Recently visited" headers don't quite match in the 'look' of the original Metropolis Semibold, and there lies the problem with the mismatch on what we are seeing and what was expected when you look at Roboto with 500 font weight in https://fonts.google.com/specimen/Roboto.

Comment (gabrielluong) [reply app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:32]: In this case, it doesn't appear as if "Jump back in" is even bolded. I am testing this on a Pixel 4.

Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/compose/SectionHeader.kt:32]: Yes, I am currently working on a workaround. I will post the results below.

File: app/src/main/res/values/styles.xml
Comment (mcarare) [inline app/src/main/res/values/styles.xml:393]: @gabrielluong For reference, this is the workaround mentioned in [this](https://github.com/mozilla-mobile/fenix/pull/22844#issuecomment-1030246430) comment, that produces the results in the attached screenshots.

Comment (gabrielluong) [reply app/src/main/res/values/styles.xml:393]: Seeing the "NewApi" - I think we should just make sure we test for Android 6 since that was the last problematic version I had to deal with.

**Comment by gabrielluong** [comment]
@mcarare It's not obvious to me when this is ready for review. So, just let me know when you want me to take a look.

**Comment by mcarare** [comment]
@gabrielluong This is ready for review. I am also running it on several devices these past days to see if something pops up as not being quite right. But maybe after we have it in nightly UX can also take a look.

File: app/src/main/res/layout/fragment_tracking_protection_blocking.xml
Comment (gabrielluong) [inline app/src/main/res/layout/fragment_tracking_protection_blocking.xml:17]: I don't see the fontWeight doing anything here. Perhaps we need to set the textStyle to bold for all of metropolis_bold.

**Review by gabrielluong — COMMENTED** [review]
I am still auditing and comparing the fonts. I do have some concerns that maybe `app:fontWeight` is kinda doing nothing. I am not seeing any changes in the font weight when I am changing the value. Maybe that's not too concerning for medium and semibold, but for bold it is definitely noticeably different.

**Comment by mcarare** [comment]
For easier tracking and reviewing, these are the transformations made:

in layout `android:fontFamily="@font/metropolis_bold" ` ->   in layout: `android:textStyle="bold"`

in layout `android:fontFamily="@font/metropolis_semibold" `  &  in layout `android:fontFamily="@font/metropolis_medium" ` ->  in styles :       
`               <item name="android:textFontWeight" tools:ignore="NewApi">@integer/font_weight_medium</item> `
`               <item name="android:fontWeight" tools:ignore="NewApi">@integer/font_weight_medium</item> `

compose `fontFamily = FontFamily(Font(R.font.metropolis_semibold))` -> `fontWeight = FontWeight.Medium` (w500)
compose `fontFamily = FontFamily(Font(R.font.metropolis_medium))` -> `fontWeight = FontWeight.Medium` (w500)

File: app/src/main/res/layout/no_collections_message.xml
Comment (gabrielluong) [inline app/src/main/res/layout/no_collections_message.xml:24]: Same question as above

File: app/src/main/res/layout/pbm_shortcut_popup.xml
Comment (gabrielluong) [inline app/src/main/res/layout/pbm_shortcut_popup.xml:36]: We can remove the textSize here

File: app/src/main/res/layout/search_suggestions_hint.xml
Comment (gabrielluong) [inline app/src/main/res/layout/search_suggestions_hint.xml:92]: Same questions as above regarding whether or not we should add the font weight

File: app/src/main/res/layout/studies_section_item.xml
Comment (gabrielluong) [inline app/src/main/res/layout/studies_section_item.xml:27]: Same as above.

File: app/src/main/java/org/mozilla/fenix/tabstray/inactivetabs/InactiveTabs.kt
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/tabstray/inactivetabs/InactiveTabs.kt:193]: I agree, and I added `headline8` here. It can be changed in a follow-up if needed.

File: app/src/main/res/layout/no_collections_message.xml
Comment (mcarare) [reply app/src/main/res/layout/no_collections_message.xml:24]: Applied `HeaderTextStyle` contains the font weight.

File: app/src/main/res/layout/fragment_sign_out.xml
Comment (mcarare) [reply app/src/main/res/layout/fragment_sign_out.xml:38]: This already had `bold` style applied, and it seemed to match better with the layout. So I just removed the font here.

File: app/src/main/res/layout/studies_section_item.xml
Comment (mcarare) [reply app/src/main/res/layout/studies_section_item.xml:27]: Same as others: this already had bold style applied, and it seemed to match better with the layout. So I just removed the font here.

File: app/src/main/res/layout/search_suggestions_hint.xml
Comment (mcarare) [reply app/src/main/res/layout/search_suggestions_hint.xml:92]: Same as others: this already had bold style applied, and it seemed to match better with the layout. So I just removed the font here.

File: app/src/main/res/layout/component_collection_creation.xml
Comment (mcarare) [reply app/src/main/res/layout/component_collection_creation.xml:183]: The link also points to snackbar design. For all places where bold style was applied I did not add font weight because it would change the current text appearance.

File: app/src/main/res/layout/component_tracking_protection_panel.xml
Comment (mcarare) [reply app/src/main/res/layout/component_tracking_protection_panel.xml:41]: I do not think the previews don't work well, I tried with different font weights, and they show up as expected only after running the app.

**Review by gabrielluong — APPROVED** [review]

**Pull request merged**
